bash/utility_functions.bash

```diff
@@ -589,4 +589,17 @@ function Make_Functions_Defined_In_This_File_Readonly()
     fi
 }
 
+: << 'DOCSTRING'
+--8<-- [start:Copy_Associative_Array-desc]
+Copy an associative array
+DOCSTRING
+function Copy_Associative_Array()
```

Comment:
Where is this used? I probably missed it. I am asking because if we decide to offer this function we need to
* document it better, in particular the interface wanting to receive two names (a calling example should be added);
* ensure the passed names refer to associative arrays
* declare `key` as local variable and spell `source` and `destination` entirely.

Reply:
This function was used at some point during the changes but has been dropped again, thanks for catching this.